crates/ruma-client-api/src/rendezvous/create_rendezvous_session.rs

```diff
@@ -0,0 +1,72 @@
+//! `POST /_matrix/client/*/rendezvous/`
+//!
+//! Create a rendezvous session.
+
+pub mod unstable {
+    //! `msc4108` ([MSC])
+    //!
+    //! [MSC]: https://github.com/matrix-org/matrix-spec-proposals/pull/4108
+
+    use http::header::{CONTENT_TYPE, ETAG, EXPIRES, LAST_MODIFIED};
+    use ruma_common::{
+        api::{request, response, Metadata},
+        metadata,
+    };
+    use url::Url;
+
+    const METADATA: Metadata = metadata! {
+        method: POST,
+        rate_limited: true,
+        authentication: None,
+        history: {
+            unstable => "/_matrix/client/unstable/org.matrix.msc4108/rendezvous",
+        }
+    };
+
+    /// Request type for the `POST` `rendezvous` endpoint.
+    #[request]
+    pub struct Request {
+        /// Any data up to maximum size allowed by the server.
+        #[ruma_api(body)]
+        pub content: String,
+
+        /// The content type, must be of the `text/plain` type.
+        #[ruma_api(header = CONTENT_TYPE)]
+        pub content_type: String,
```

Comment:
Since the content type cannot change, I don't think it makes sense as its own field. Right now it requires manual implementation to add it though. Besides, it looks like we also need to add `Content-Length`, which should be computed from the content.

Reply:
`content-length` is not something we should concern ourselves with, hyper adds it automatically for outgoing requests.

Reply:
> `content-length` is not something we should concern ourselves with, hyper adds it automatically for outgoing requests.

Hmm, this doesn't seem to work if I post an empty string as the content content, which makes sense since there's no content. On the other hand Synapse implementation complains about the `content-length` not being present.

Any opinions on what to do? To make it work I started adding the `content-length` manually.

Reply:
I'd call that a Synapse bug, generally `content-length` absensce means that the body is empty...

Reply:
> I'd call that a Synapse bug, generally content-length absensce means that the body is empty... Though I'd not be strongly opposed to setting it here if it's part of the docs specifically for this endpoint and not hard to implement (should be trivial if the body is a String / Vec<u8>).

That's what I implemented in https://github.com/ruma/ruma/pull/1800/commits/15d37af504768d9cc5c83d50a555a5162830f828. I guess I didn't add a doc comment calling this out.